plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/data/managers/AbstractTextPanelEditor.java

```diff
@@ -299,11 +377,27 @@ public void primeEditorValue(@NotNull DBRProgressMonitor monitor, @NotNull Style
                     }
                     applyEditorStyle();
                 }
+            }
+        });
+    }
+
+    private void showRestrictedContent(@NotNull DBDContent value) throws DBCException, IOException {
```

Comment:
Let's also rename it as "restricted" doesn't quite fit the purpose of this method. Let's call it something like "showLimitedContent", you get the spirit.

Also, let's pass `maxContentSize` as an argument (already computed by the caller) to avoid code duplication.

Reply:
Done